src/backend/db/feature-permissions.js

```diff
@@ -0,0 +1,219 @@
+        ON perms.group_id = group_ids.group_id;
+    `;
+
+    let resp = (await db.raw(query, {userId: userId})).rows;
+    let allowedActions = resp.map(row => row.feature_name);
+    let groupIds = resp.map(row => row.group_id);
+    return {
+      allowedActions: [...new Set(allowedActions)], // return only unique values (remove duplicates). See https://stackoverflow.com/questions/1960473/get-all-unique-values-in-a-javascript-array-remove-duplicates
+      groupIds: [...new Set(groupIds)], // return only unique values (remove duplicates)
+    };
+  } catch (err) {
+    console.log('[getAllowedActions] ERROR: ', err);
+    throw err;
+  }
+};
+
+/**
+ * Sync the locally-stored user/group data with up-to-date data from Google Groups
+ * 
+ * @param {Array[String]} groupIds
+ * @param {String} accessToken: Oauth2 Access token
+ * 
+ * @return [nothing]
+ */
+const updateUserGroups = (db) => async (userId, groupIds, accessToken) => {
+
+  let gAdminClient;
+  let newGroups;
+  try {
+    gAdminClient = getGAdminClient(accessToken);
+
+    // Remove user from old groups
+    await db.raw(`DELETE FROM group_users WHERE user_id = :userId;`, {userId: userId});
+    // Convert a list of group ids to a list of {group ids & group keys}
+    newGroups = await appendKeysToIds(db, groupIds);
+    // console.log("newGroups")
+    // console.log(newGroups)
+  } catch(err) {
+    console.error(err);
+    throw err;
+  }
+    
+  for (let g = 0; g < newGroups.length; g++) {
+    const {id: groupId, group_key: groupKey} = newGroups[g];
+    try {
+      // Get up-to-date information for the current group (membership information) from Google Groups
+      let resultList = await gAdminClient.members.list({ 
+          groupKey: groupKey
+      });
+
+      db.transaction(function(trx) {
+        db('group_users')
+          .where('group_id', groupId)  
+          .del()
+          .transacting(trx)
+          .then(function() {
+            // Fill the group_users table with up-to-date membership info.
+            const memberIds = resultList.data.members.map(member => member.id);
+            // console.log("memberIds");
+            // console.log(memberIds);
+
+            if (memberIds && memberIds.length > 0) {
+              return db('group_users')
+                .insert(memberIds.map(m_Id => { return {user_id: m_Id, group_id: groupId}} ))
+                .transacting(trx);
+            }
+          })
+          .then(trx.commit)
+          .catch(trx.rollback);
+      })
+      .then(function() {
+        console.log("Successfully updated Google Groups membership info for group: " + groupKey);
+      })
+      .catch(async function(error) {
+        console.error(`Error: Failed to update group_users table for group: ${groupKey}. `, error); 
+        console.log('Skipping Group: ' + groupKey);
+        try {
+          await db.raw(`INSERT INTO group_users (user_id, group_id) VALUES (:userId, :groupId);`, {userId: userId, groupId: groupId});
```

Comment:
This is copied from CMS right ? I don't think this line is needed actually. 

I think this was intended to manually insert the deleted line back in case one of the previous DB Delete queries failed. But since they are wrapped in a transaction, I don't think this is needed anymore.

Can you please test and investigate this (make sure it can be removed) ?

If that's too much effort right now, please create a separate ticket for this, thanks !